tests/__init__.py

```diff
@@ -829,6 +829,14 @@ def wrapper(wrapped, instance, args, kwargs):
         # identify the snapshot.
         token = "{}{}{}.{}".format(module.__name__, "." if clsname else "", clsname, wrapped.__name__)
 
+        # Use variant that applies to update test token. One must apply. If none
+        # apply, the test should have been marked as skipped.
+        if variants:
+            applicable_variant_ids = [k for (k, v) in variants.items() if v is True]
+            assert len(applicable_variant_ids) == 1
+            variant_id = applicable_variant_ids[0]
+            token = "{}_{}".format(token, variant_id) if variant_id else token
```

Comment:
Is this check actually necessary? It should always be a string (which should be safe to format with). We should maybe do some validation on the id as well since it has to be a valid file name.

Reply:
The check was intended to avoid updating the token if the variant_id is `None` or `""`.

Reply:
ah true right. We wouldn't want the underscore in those cases. :+1: